### PR TITLE
sync engine tokens data

### LINKED ISSUE #2593: selected tokens sync using posting json metadata.
The goal is to sync selected tokens/assets between mobile and web, later potentially among various hive apps.

- user can update selected tokens using `assetSelectModal.tsx` accessible from wallet screen.
- format selected tokens/assets according to format give in reference link
```
{
  "profile": {
      "tokens": {"engine":["LEO","BEE"],"spk": ["SPK", "LARYNX", "LP"]},
      "name":"xyz",
       ...,
      "version": 2}
}
```
- use `profileUpdate()` method from`dhive.js `to update profile
- update selected tokens cache in redux from `account.currentAccount.about.profile.tokens`
- keep using redux for now, later we can use react-query if required.
- sync `about.profile.tokens` with wallet.selectedCoins `walletScreen.tsx` currentAccount change side effect 
`
ref: https://discord.com/channels/@me/849624730234781707/1065886783436509254

### PR DESCRIPTION
**Pull request by aliseyalvi**
### What does this PR?
This PR update profile meta with selected engine tokens data and updates the selected tokens from profile json meta

### Issue number
fixes #2593

### CODE REVIEW

File: src/screens/wallet/screen/walletScreen.tsx [hunk 1]
```diff
@@ -91,31 +77,52 @@ const WalletScreen = ({ navigation }) => {
...
+    }));
+  };
 
+  const _updateSelectedCoinsDataFromProfileJsonMeta = () => {
+    const currSelectedEngineTokens = selectedCoins.filter(
+      (item) => !DEFAULT_ASSETS.some((defaultCoin) => defaultCoin.id === item.id),
+    );
+    if (
+      currentAccount.about?.profile?.tokens &&
+      currentAccount.about?.profile?.tokens?.engine.length !== currSelectedEngineTokens.length // check if current selected engine tokens differ from profile json meta
```
Comment (noumantahir) [inline src/screens/wallet/screen/walletScreen.tsx:103]: not a good comparison logic...see below example invalidating this logic... instead stringify might be a good option here.

Screenshot 2023-01-24 at 5 28 43 PM

File: src/screens/wallet/children/assetsSelectModal.tsx [hunk 2]
```diff
@@ -9,13 +9,18 @@ import { CoinBase, CoinData } from '../../../redux/reducers/walletReducer';
...
   const dispatch = useAppDispatch();
   const intl = useIntl();
 
   const coinsData = useAppSelector((state) => state.wallet.coinsData);
   const selectedCoins: CoinBase[] = useAppSelector((state) => state.wallet.selectedCoins);
+  const pinCode = useAppSelector((state) => state.application.pin);
+  const currentAccount = useAppSelector((state) => state.account.currentAccount);
 
   const [visible, setVisible] = useState(false);
   const [selection, setSelection] = useState(selectedCoins.filter((item) => item.isEngine));
```
Comment (noumantahir) [inline src/screens/wallet/children/assetsSelectModal.tsx:26]: selection not syncing with updated redux values, for quick fix reset selection using `setSelection()` in `showModal()`

https://user-images.githubusercontent.com/6298342/214293072-f5bdc093-d2bb-419d-9b38-699eb0703d96.mov

**Review by noumantahir — CHANGES_REQUESTED** [review]
Two changes required...

- refine logic for updating selected coins
- update selection on showModal call

Generally try to use `Asset` instead of `Coin`, initially wallet was created with coins in mind, now that we have tokens, Asset seems like a good general phrase, later I will also update redux but since it's already in production, that part will require being extra careful

**Review by noumantahir — APPROVED** [review]
seems good

**Review by feruzm — APPROVED** [review]
Lgtm

**Pull request merged**